dace/sdfg/validation.py

```diff
@@ -370,8 +466,26 @@ def validate_state(state: 'dace.sdfg.SDFGState',
                 )
         ########################################
 
+    #Check that HBM Magic Indices are evaluatable
+
+        def validate_hbm_subset(magic_subset: subsets.Subset):
```

Comment:
Since this function is called `validate`, why not throw an exception from here directly?

Also, please get rid of "magic" here (and everywhere in your code, just search for "magic" and make sure nothing is left).